drake/systems/plants/@RigidBodyManipulator/doKinematics.m

```diff
@@ -88,8 +88,17 @@
 % gradient computations.
 kinsol.q = q;
 kinsol.v = v;
+kinsol.has_gradients = options.compute_gradients;
 
-if (options.use_mex && model.mex_model_ptr~=0 && isnumeric(q))
+if options.use_mex && model.mex_model_ptr~=0 && ((isnumeric(q) && isnumeric(v)) || (isa(q, 'TaylorVar') && getOrder(q) < 2)) % relying on short circuiting here
+  if ~isnumeric(q)
+    if isa(q, 'TaylorVar')
+      if options.compute_gradients
+        error('can only compute first order gradients using mex')
```

Comment:
Handling all of the possible inputs in doKinematics while trying to be efficient took a while to get right; I don't like how complicated this logic is currently...

Yes, I am indeed trying to avoid that case. I'll make the message more descriptive.